core-bundle/contao/themes/flexible/styles/basic.css

```diff
@@ -405,6 +405,15 @@ table.with-zebra tbody tr:nth-child(even) td {
 	display: contents;
 }
 
+/* Legend buttons */
+button[data-action="contao--toggle-fieldset#toggle"] {
+	all: unset;
+}
+
+button[data-action="contao--toggle-fieldset#toggle"]:focus-visible {
+	outline: -webkit-focus-ring-color auto 1px;
+}
```

Comment:
I think instead of adding this here, we should change the `fieldset.tl_tbox > legend, fieldset.tl_box > legend` styles in the main.css to `fieldset.tl_tbox > legend > button, fieldset.tl_box > legend > button`.
With your current solution a click on the arrow-symbol would probably not trigger the action, right?

Reply:
And I think `-webkit-focus-ring-color` is not supported everywhere. Maybe we need a fallback for that?

Reply:
> I think instead of adding this here, we should change the `fieldset.tl_tbox > legend, fieldset.tl_box > legend` styles in the main.css to `fieldset.tl_tbox > legend > button, fieldset.tl_box > legend > button`. With your current solution a click on the arrow-symbol would probably not trigger the action, right?

Good catch, might want to restyle the whole button to full-width then with the `:has`-selector due to aschempp's comment in https://github.com/contao/contao/pull/7235#pullrequestreview-2080925161.

> And I think `-webkit-focus-ring-color` is not supported everywhere. Maybe we need a fallback for that?
See my comment above, I'd rather use proper styles now than resetting since handling any browser-native styles across all browsers is kinda a PITA, especially with webkit / non-webkit / safari

Reply:
You do realize that we don‘t have to use buttons at all costs, right? We can use `aria-role="button"` and `aria-pressed` on non-button elements to make them accessible.

<img width="775" alt="" src="https://github.com/contao/contao/assets/1192057/15ce6330-c907-4ab1-bc13-960d69c19652">

Reply:
That would still not make it focusable @leofeyer.

If not going for a native `<button>`, you'd still have to use the following markup to achieve the same result and to make it accessible which is what could be avoided by going for the native HTML solution.

**Not using a native button (Also adding the `aria-pressed` state via Stimulus)**
```html
<legend tabindex="0" aria-role="button" aria-pressed="false" data-action="click->contao--toggle-fieldset#toggle keydown.enter->contao--toggle-fieldset#toggle keyup.space->contao--toggle-fieldset#toggle:prevent">My legend</legend>
```

**Native button**
```html
<legend><button type="button" data-action="contao--toggle-fieldset#toggle">My legend</button></legend>
```

Reply:
AFAIK the best practice accessibility-wise is to use a `<button>` for such things instead of implementing all the accessibility features yourself (focus, keyboard navigation, ARIA role,…)

But I agree we should add `aria-pressed="…"` to the button to improve the accessibility even more.

Reply:
> We may also leverage "aria-expanded" now but I am unsure if that is necessary

Probably the same here if we are talking about screen readers.

Reply:
Since it indicates that the `<button>` becomes a "toggle" button, it would further enhance it:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed

It's not redundant since buttons aren't just for toggle states.

Reply:
I restyled it within the last commits  a3c847d31dd894a69a76b940a05c7bcb02e9ad08 29d597edda73194519a9dff7c8f1b1aed7b16c0b

We still have to add the states within the stimulus controller